tachyon/math/polynomials/univariate/naive_batch_fft.h

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2022 The Plonky3 Authors
+// Use of this source code is governed by a MIT/Apache-2.0 style license that
+// can be found in the LICENSE-MIT.plonky3 and the LICENCE-APACHE.plonky3
+// file.
+
+#ifndef TACHYON_MATH_POLYNOMIALS_UNIVARIATE_NAIVE_BATCH_FFT_H_
+#define TACHYON_MATH_POLYNOMIALS_UNIVARIATE_NAIVE_BATCH_FFT_H_
+
+#include <optional>
+#include <vector>
+
+#include "tachyon/base/bits.h"
+
+namespace tachyon::math {
+
+template <typename F>
+class NaiveBatchFFT {
+ public:
+  using Config = typename FiniteFieldTraits<F>::Config;
+
+  // Compute the discrete Fourier transform (DFT) of each column in `mat`.
+  constexpr static void DFTBatch(RowMajorMatrix<F>& mat) {
+    size_t rows = static_cast<size_t>(mat.rows());
+    size_t cols = static_cast<size_t>(mat.cols());
+    CHECK(base::bits::IsPowerOfTwo(rows));
+    F g;
+    CHECK(F::GetRootOfUnity(rows, &g));
+
+    RowMajorMatrix<F> res = RowMajorMatrix<F>::Constant(rows, cols, F::Zero());
+
+    std::vector<F> points = F::GetSuccessivePowers(rows, g);
+    size_t num_points = points.size();
+
+    for (size_t res_r = 0; res_r < num_points; ++res_r) {
+      std::vector<F> point_powers = F::GetSuccessivePowers(rows, points[res_r]);
+      for (size_t src_r = 0; src_r < num_points; ++src_r) {
+        for (size_t col = 0; col < cols; ++col) {
+          res(res_r, col) += point_powers[src_r] * mat(src_r, col);
+        }
+      }
+    }
+    mat = res;
+  }
+
+  // Compute the inverse DFT of each column in `mat`.
```

Comment:
```suggestion
  // Compute the inverse DFT of each column in |mat|.
```
Let's follow the c++ style.